fix: Type error with YAML dates in frontmatter

jgraichen opened the pull request:
When valid YAML dates or times are added to the page frontmatter, the YAML parser will already convert them to `datetime.date` or `datetime.datetime` objects.

Example:

    ---
    date: 2023-04-12
    ---

This resulted in #strptime raising a `TypeError` because `datetime.date` isn't a string. This commit changes the plugin code to always accept a native `datetime.date` or `datetime.datetime`, even if `meta_time_format` is not set, and handle them without string parsing.

Close #48

jgraichen commented on mkdocs_blogging_plugin/plugin.py:
```diff
@@ -337,9 +337,9 @@ def with_timestamp(self, page, by_revision):
+        elif "date" in page.meta:
             timestamp = self._parse_time(page.meta["date"])
-        if not timestamp:
+        else:
```
If `time` is included in the page meta but is _not_ a date time, and `meta_time_format` is not set, this will end up with `timestamp = None`.

For example:

```
---
time: 5 minutes
date: 2024-04-12
---
```

The `time` meta could be meant for another plugin.

Using `is None` would try with `"date"` if `"time"` is present but cannot be parsed.

liang2kl replied:
Okay, I understand. Thank you for the explanation.

liang2kl commented:
By the way, could you please update the docs? The related stuff should be in the "Global Settings" section of `index.md`.

jgraichen commented:
I added an extra test about the fallback if `time:` is invalid and some more explicit return paths for the `_parse_time` method.

> By the way, could you please update the docs? The related stuff should be in the "Global Settings" section of `index.md`.

I've added a note.
